doc/classes/Parallax2D.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="Parallax2D" inherits="Node2D" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		A node used to create a parallax scrolling background.
+	</brief_description>
+	<description>
+		A Parallax2D is used to create a parallax effect. It can move at a different speed relative to the camera movement using [member scroll_scale]. This creates an illusion of depth in a 2D game. If manual scrolling is desired, the [Camera2D] position can be ignored with [member override_ignore_camera_scroll].
+		[b]Note:[/b] Any changes to this node's position made after it enters the scene will be ignored.
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="autoscroll" type="Vector2" setter="set_autoscroll" getter="get_autoscroll" default="Vector2(0, 0)">
+			When set, the offset will scroll automatically by the set amount of pixels per second.
+		</member>
+		<member name="limit_begin" type="Vector2" setter="set_limit_begin" getter="get_limit_begin" default="Vector2(0, 0)">
+			Top-left limits for scrolling to begin. If the camera is outside of this limit, the Parallax2D will stop scrolling. Must be lower than [member limit_end] to work.
+		</member>
+		<member name="limit_end" type="Vector2" setter="set_limit_end" getter="get_limit_end" default="Vector2(0, 0)">
+			Bottom-right limits for scrolling to end. If the camera is outside of this limit, the Parallax2D will stop scrolling. Must be higher than [member limit_begin] to work.
+		</member>
+		<member name="offset" type="Vector2" setter="set_offset" getter="get_offset" default="Vector2(0, 0)">
+			The Parallax2D's offset relative to its parent.
+		</member>
+		<member name="override_follow_viewport" type="bool" setter="set_follow_viewport" getter="get_follow_viewport" default="true">
+			If [code]true[/code], Parllax2D will follow along with the current camera. It may be desired to turn off if the Parallax2D is in a [CanvasLayer] separate from the current camera.
+		</member>
+		<member name="override_ignore_camera_scroll" type="bool" setter="set_ignore_camera_scroll" getter="is_ignore_camera_scroll" default="false">
+			If [code]true[/code], Parallax2D's position will be affected by the position of the camera.
+		</member>
+		<member name="override_repeat_times" type="int" setter="set_repeat_times" getter="get_repeat_times" default="1">
```

Comment:
It's weird that it's an override not a normal property like `repeat_times`. I assume texture repeats are some internal property in RenderindServer?

Reply:
Yep! Before this, textures could only repeat once. I've modified the renderer to repeat based on this amount now, since it was only used for parallax in the first place. Since this shouldn't be a common use case (it wasn't handled at all by `ParallaxBackground`) I thought it made more sense to put this is in the override section since it should only be used if necessary.

Reply:
Well personally I always found the ParallaxBackground's behavior very limiting. It forced you to use very big images to  cover the entire screen.

Reply:
Yeah, but I'm not sure if there's really a way around that without limiting it in other ways. For infinite repeat, it'll need to be larger than the screen no matter what, but at least with this, you can choose to repeat at the texture or for the entire parallax layer. This way you can have small images repeated by the screen, repeat them by the texture/region_rect until they're big enough to be continuous when repeated by the screen, or just use a large enough image that's meant to be continuous to start with to repeat by the screen. This is more to overcome the issue with `ParallaxBackground` where zooming out shows that there's only two repeats shown at any given time.

I think there's still some editor helpers and stuff like we discussed in the other ticket that may be good to improve this, but I think this should make for a better groundwork to build off of.